datafusion/core/src/execution/context/mod.rs

```diff
@@ -859,7 +859,7 @@ impl SessionContext {
         // check if the file extension matches the expected extension
         for path in &table_paths {
             let file_name = path.prefix().filename().unwrap_or_default();
-            if !path.as_str().ends_with(&option_extension) && file_name.contains('.') {
+            if !file_name.ends_with(&option_extension) && file_name.contains('.') {
```

Comment:
```suggestion
            if !file_name.ends_with(&option_extension) && !path.is_collection() {
```
Perhaps?

Reply:
```suggestion
            if (!path.as_str().ends_with(option_extension.clone().as_str())
                && !path.is_collection())
                || (!path
                    .as_str()
                    .ends_with((option_extension.clone() + "/").as_str())
                    && path.is_collection())
           {
```
If the path ends with a forward slash `/`, then check the extension with another forward slash `/`. Otherwise, simply check the extension.

Reply:
That doesn't compile for me. Perhaps this method isn't available in the version of arrow-rs that we are currently using? 

```
no method named `is_collection` found for reference `&ListingTableUrl` in the current scope
```

Reply:
`ListingTableUrl` is a DataFusion concept, perhaps your checkout is behind main?